common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java

```diff
@@ -0,0 +1,35 @@
+package com.microsoft.identity.common.internal.providers.oauth2;
+
+import com.microsoft.identity.common.Account;
+import android.net.Uri;
+
+/**
+ * Created by shoatman on 11/15/2017.
+ */
+
+public abstract class OAuth2Strategy {
+
+    protected String mTokenEndpoint;
+    protected String mAuthorizationEndpoint;
+    protected Uri mIssuer;
+
+    protected AuthorizationResponse requestAuthorization(AuthorizationRequest request, AuthorizationStrategy authorizationStrategy){
+        validateAuthoriztionRequest(request);
+        Uri authorizationUri = createAuthorizationUri();
+        AuthorizationResult result = authorizationStrategy.requestAuthorization(request);
+        //TODO: Reconcile authorization result and response
```

Comment:
In this case the AuthorizationResult is specific to OAuth/OIDC... we may want to make that more clear in the name of the class instead of relying on the package name to indicate that.  I think exposing the response as part of the result probably does make sense.

I haven't add it yet, but I plan to introduce a "controller" class for ADAL... and for MSAL to map adapt the public API to the common underlying API structure... and to act as a coordinator of the request.  (In the case of ADAL that coordination is tricky....)